module_utils/network_lsr/argument_validator.py

```diff
@@ -320,6 +322,14 @@ def _validate_impl(self, value, name):
         result = []
         for (idx, v) in enumerate(value):
             try:
+                if self.enum_values is not None and not any(
+                    re.match(str(x), str(v)) for x in self.enum_values
```

Comment:
the conversion to `str()` doesn't seem right. If the values are not strings to being with, it's not clear how the pattern matching should work (it should not work at all).

Reply:
updated

Reply:
`str()` removed